R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
In some cases, `data` is not the full dataset and hence does not contain `x` and `y` but just data with subsetted columns.

Set a breakpoint at `if (check.data)` in `makeSupervisedTask()` in the following example. You will see how `data` changes. I need `x` and `y` because I use them to check if the the task `is.spatial`. If they are not present due to some subsetting, `makeSupervisedTask()` will report `is.spatial = FALSE` although it should be `TRUE`. 

```r
data(basque, package = "sperrorest")
basque$geometry <- NULL
task <- makeClassifTask(id = "pathogen_data", data = basque, 
                        target = "diplo01", positive = "1",
                        spatial = TRUE)
```

So there was no bug up to now - I just need always need the full dataset here for my further checks.